[billieboy7} iP

billieboy7 opened the pull request:
# Duke - Task Tracker

> "Implement whatever organizational system works best for you, but then be sure to review your to-do list at least once a week." – Cheryl Barker ([Source](https://bookroo.com/quotes/to-do-lists))

Liberate your mind from the burden of remembering tasks. Duke offers:

- Easy to use task tracker
- Intuitive text-based interface
- Remarkable ~~speed~~ SPEED


So what do you have to do :
1. Download it from [here](https://github.com/billieboy7/ip/releases/tag/A-Jar).
2. Double-click the executable.
3. Input your tasks.
4. Let it handle your task management for you 😉


**Awesome Features**
- [x] Task management
- [ ] Calendar View (coming soon)
- [ ] Reminders (coming soon)



If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:

```java
    public static void main(String[] args) {

        Duke duke = new Duke("./data/data.txt");
        duke.start();

    }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

billieboy7 committed:
test

billieboy7 committed:
Level-0

billieboy7 committed:
Level-1

billieboy7 committed:
Level-2

billieboy7 committed:
Level-3

billieboy7 committed:
Level-4

billieboy7 committed:
Level-5

billieboy7 committed:
Level-6

billieboy7 committed:
Level-6

billieboy7 committed:
test

billieboy7 committed:
branch-Level-7

billieboy7 committed:
Merge branch 'branch-Level-7'

billieboy7 committed:
branch-Level-8

billieboy7 committed:
Merge branch 'branch-Level-8'

billieboy7 committed:
Implemented Storage class

billieboy7 committed:
Implemented most of UI class

billieboy7 committed:
Command class implemented. Yet to solve IO exception handling

billieboy7 committed:
Yet to Solve Mark/Unmark - Save changes issue

billieboy7 committed:
IO exception resolved

billieboy7 committed:
Merge branch 'A-MoreOOP'

billieboy7 committed:
branch-A-Packages

billieboy7 committed:
Merge branch 'A-Packages'

billieboy7 committed:
Merge branch 'add-gradle-support' into fetch-gradle

billieboy7 committed:
Merge branch 'fetch-gradle'

billieboy7 committed:
A-Gradle

billieboy7 committed:
Merge branch 'A-Gradle'

billieboy7 committed:
branch-A-Junit

billieboy7 committed:
Merge branch 'A-Junit'

billieboy7 committed:
Solve coding standard issues on a separate branch of master

billieboy7 committed:
Continue editing from ByeCommand

billieboy7 committed:
done with Coding Standard. Check for compliance in future versions

m1oojv commented on src/main/java/Exception/DukeException.java:
Names representing packages should be in all lower case. I think this mistake also happened across a few of your other packages too.

`exception.DukeException`

m1oojv reviewed (changes requested):
Overall, I found your code easy to read for the most part. I noted a few minor coding standard violations.

jamz903 commented on src/main/java/Task/Task.java:
```diff
@@ -0,0 +1,28 @@
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public void taskDone(boolean status) {
```
Minor coding violation! Boolean variables/methods should be named to sound like booleans, and setter methods for boolean variables should be in the following form instead:
```suggestion
    public void setDone(boolean isDone) {
```

jamz903 reviewed (commented):
Overall, I found your code easy to read for the most part, except that your code could use a bit of documentation for easier readability. I noted a minor coding violation, but otherwise your code is very well done 👍

jamz903 commented on src/main/java/Command/ByeCommand.java:
```diff
@@ -0,0 +1,22 @@
+    }
+
+    @Override
+    public boolean end() {
```
```suggestion
    public boolean getEnd() {
```

jamz903 commented on src/main/java/Command/Command.java:
```diff
@@ -0,0 +1,16 @@
+
+    public void execute(TaskList taskList, UI ui, Storage storage) throws DukeException {}
+
+    public boolean end() {
```
```suggestion
    public boolean getEnd() {
```

jamz903 reviewed (commented):
Noted another minor coding violation, that has to do with one of your methods!

billieboy7 committed:
done with find feature

billieboy7 committed:
Merge branch 'branch-A-JavaDoc'

billieboy7 committed:
Merge branch 'branch-A-CodingStandard'

billieboy7 committed:
A-CodingStandard

billieboy7 committed:
Merge branch 'branch-Level-9'

billieboy7 committed:
add Find feature

billieboy7 committed:
Update README.md

billieboy7 committed:
Clean up UI usage

billieboy7 committed:
Add GUI

billieboy7 committed:
Merge branch 'branch-Level-10'

billieboy7 committed:
Insert assert statements
There are no checks currently on how the user input is being handled within the program itself.

Using assert statements checks whether the input is being handled properly within the programme and whether each method is working in the way it is intended to.

Assert statements are preferred because their less disruptive.

billieboy7 committed:
ListCommand: updated package names and return type for execute method

Lack of standardisation of package names threw errors. execute method did not return anything which the gui package could use.

Standardised names of the package imports were from. execute method returns string to display on gui.

billieboy7 committed:
Merge pull request #2 from billieboy7/branch-A-Assertions

Insert assert statements

billieboy7 committed:
Merge branch 'master' into branch-A-CodeQuality

billieboy7 committed:
Merge pull request #3 from billieboy7/branch-A-CodeQuality

Merge branch 'master' into branch-A-CodeQuality

billieboy7 committed:
Parser class: abstract away the validation of commands into the parsing methods

Parser class initially had 2 different operations - 1) To decide the type of command to be returned 2) Check for the validity of the formatting of the command once the type of command to be returned was determined.

Abstracting the second operation(validation check) allows the parse method to be simpler and look neater.

billieboy7 committed:
Add update description functionality

Users cant change the description without deleting the task.

Introduced a new functionality that allows for the updating of the tasks without deleting the older tasks.

Makes it easier, convenient and faster for the updating of task descriptions especially when there are multiple changes.

billieboy7 committed:
Add feature: Dialog Box does not truncate because of large output [GUI]

billieboy7 committed:
Add User Guide in README.md

billieboy7 committed:
Merge branch 'branch-A-UserGuide'

billieboy7 committed:
Add Ui.png

billieboy7 committed:
Update javadoc comments for public methods

billieboy7 committed:
Rectify out of index user input handling for delete, mark, unmark

billieboy7 committed:
Rectify Header Formatting for README.md

billieboy7 committed:
Change README.md header
